Forms/MainForm.cs

```diff
@@ -129,6 +131,15 @@ private void bRandomise_Click(object sender, EventArgs e)
             bgWorker.RunWorkerAsync();
         }
 
+        private void bApplyPatch_Click(object sender, EventArgs e)
+        {
+            if (openPatch.ShowDialog() == DialogResult.OK)
+            {
+                _settings.InputPatchFilename = openPatch.FileName;
+                bRandomise_Click(sender, e);

```

Comment:
Minor; I'd rather not call the click function here, but instead have both the `bRandomise_Click()` and `bApplyPatch_Click()` call a common function.